### ISSUE TITLE
[YSQL] Better message to be given when node using tablespace is not available (Wrong number of responses: 0, while 5 expected: 0 vs 5)

### ISSUE THREAD

**Issue by shamanthchandra-yb** [yugabyte/yugabyte-db#17042]
Jira Link: [DB-6353](https://yugabyte.atlassian.net/browse/DB-6353)
### Description

**Scenario**

1. Create table space like this:

```

CREATE TABLESPACE tablespace0 WITH (
  replica_placement='{"num_replicas": 2, "placement_blocks":
  [{"cloud":"aws","region":"eu-central-1","zone":"eu-central-1b","min_num_replicas":1},
  {"cloud":"aws","region":"eu-central-1","zone":"eu-central-1c","min_num_replicas":1}]}'
);
CREATE TABLESPACE tablespace1 WITH (
  replica_placement='{"num_replicas": 2, "placement_blocks":
  [{"cloud":"aws","region":"sa-east-1","zone":"sa-east-1b","min_num_replicas":1},
  {"cloud":"aws","region":"sa-east-1","zone":"sa-east-1c","min_num_replicas":1}]}'
);
CREATE TABLESPACE tablespace2 WITH (
  replica_placement='{"num_replicas": 1, "placement_blocks":
  [{"cloud":"aws","region":"ap-northeast-1","zone":"ap-northeast-1c","min_num_replicas":1}]}'
); 
```
2. Create table with partitions. I had a parent table and 3 child tables, pointing each one to different table space.
3. Remove a node associated with tablespace2 (I had a universe of 5 nodes)
4. Try to insert a row pertaining to this tablespace OR try to do select * from parent table name

**Observation:**
```
yugabyte=# 
postgres=# INSERT INTO sqlgeopartitionedtable VALUES ('region_2', 'c', 'a', 1);
ERROR:  Wrong number of responses: 0, while 1 expected: 0 vs 1

postgres=# select count(*) from sqlgeopartitionedtable_2;
ERROR:  Wrong number of responses: 0, while 5 expected: 0 vs 5

postgres=# DELETE FROM sqlgeopartitionedtable;
ERROR:  Wrong number of responses: 0, while 1 expected: 0 vs 1
```

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-6353]: https://yugabyte.atlassian.net/browse/DB-6353?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by shamanthchandra-yb** [comment]
Tried on build 2.18.0.0-b64

I no longer see above error, instead seeing:

`
postgres=# INSERT INTO sqlgeopartitionedtable VALUES ('region_2', 'B', 'B', 2);
ERROR:  Perform RPC (request call id 779) to 172.160.50.101:9100 timed out after 120.000s
`

**Comment by fizaaluthra** [comment]
fixed by commit eb6fae3f3d21f18c2884e18696a4e8b1b093b9f4 (2.18 backport cdfc59351ab736968b9f09d84fc9bd655270c311)
